src/main/java/org/cirdles/topsoil/app/TopsoilMainWindow.java

```diff
@@ -323,6 +329,17 @@ private void initializeAndShow(JavaScriptChart javaScriptChart) {
             initializeAndShow(javaScriptChart, dataset);
         });
     }
+    
+    private Dataset filter(Dataset dataset)
+    {
+        Iterator<Entry> it = dataset.getEntries().iterator();
+        while(it.hasNext()) {
+            if(!it.next().getIsSelected()) {
+                it.remove();
+            }
+        }
+        return dataset;
+    }
```

Comment:
I would rather you not modify the dataset. This isn't guaranteed to work (because a `List` doesn't have to support modification; check the docs), and it doesn't encourage good code. Remember that you can add secret fields to entries.